internal/monitors/conviva/metricconfig.go

```diff
@@ -0,0 +1,223 @@
+		"conviva.quality_metriclens.video_startup_time_sec",
+		"conviva.quality_metriclens.rebuffering_ratio_percent",
+		"conviva.quality_metriclens.average_bitrate_kbps",
+		"conviva.quality_metriclens.video_playback_failures_percent",
+		"conviva.quality_metriclens.ended_plays",
+		"conviva.quality_metriclens.connection_induced_rebuffering_ratio_percent",
+		"conviva.quality_metriclens.video_restart_time",
+	},
+	"audience_metriclens": {
+		"conviva.audience_metriclens.concurrent_plays",
+		"conviva.audience_metriclens.plays",
+		"conviva.audience_metriclens.ended_plays",
+	},
+}
+
+// metricConfig for configuring individual metric
+type metricConfig struct {
+	// Conviva customer account name. The default account is fetched used if not specified.
+	Account                     string   `yaml:"account"`
+	Metric                      string   `yaml:"metric" default:"quality_metriclens"`
+	Filters                     []string `yaml:"filters"`                     // Filter names. The default is `All Traffic` filter
+	MetricLensDimensions        []string `yaml:"metricLensDimensions"`        // MetricLens dimension names. The default is names of all MetricLens dimensions of the account
+	ExcludeMetricLensDimensions []string `yaml:"excludeMetricLensDimensions"` // MetricLens dimension names to exclude.
+	accountID                   string
+	filters                     map[string]string  // id:name map of filters derived from the configured Filters
+	metricLensDimensionMap      map[string]float64 // name:id map of MetricLens dimensions derived from configured MetricLensDimensions
+	isInitialized               bool
+	filtersWarmup               map[string]string // id:name map of filters in filters_warmup status on response
+	filtersNotExist             map[string]string // id:name map of filters in filters_not_exist status on response
+	filtersIncompleteData       map[string]string // id:name map of filters in filters_incomplete_data status on response
+	mutex                       *sync.RWMutex
+}
+
+func (mc *metricConfig) init(service accountsService) {
+	mc.mutex.Lock()
+	defer mc.mutex.Unlock()
+	if !mc.isInitialized {
+		if err := mc.setAccount(service); err != nil {
+			logger.Errorf("Metric %s account setting failure. %+v", mc.Metric, err)
+			return
+		}
+		if err := mc.setFilters(service); err != nil {
+			logger.Errorf("Metric %s filter(s) setting failure. %+v", mc.Metric, err)
+			return
+		}
+		if err := mc.setMetricLensDimensions(service); err != nil {
+			logger.Errorf("Metric %s MetricLens dimension(s) setting failure. %+v", mc.Metric, err)
+			return
+		}
+		if err := mc.excludeMetricLensDimensions(service); err != nil {
+			logger.Errorf("Metric %s MetricLens dimension(s) exclusion failure. %+v", mc.Metric, err)
+			return
+		}
+		mc.isInitialized = true
+	}
+}
+
+// setting account id and default account if necessary
+func (mc *metricConfig) setAccount(service accountsService) error {
+	if strings.TrimSpace(mc.Account) == "" {
```

Comment:
If you just do the trim on line 82 above this line then you can just check for `mc.Account == ""` instead of doing the trim twice.